commitizen/config/__init__.py

```diff
@@ -3,28 +3,35 @@
 from pathlib import Path
 
 from commitizen import defaults, git
+from commitizen.exceptions import ConfigFileNotFound, ConfigFileIsEmpty
 
 from .base_config import BaseConfig
 from .json_config import JsonConfig
 from .toml_config import TomlConfig
 from .yaml_config import YAMLConfig
 
 
-def read_cfg() -> BaseConfig:
+def read_cfg(filepath: str | None = None) -> BaseConfig:
     conf = BaseConfig()
 
-    git_project_root = git.find_git_project_root()
-    cfg_search_paths = [Path(".")]
-    if git_project_root:
-        cfg_search_paths.append(git_project_root)
+    if filepath is not None:
+        cfg_paths = (path for path in (Path(filepath),))
+    else:
+        git_project_root = git.find_git_project_root()
+        cfg_search_paths = [Path(".")]
+        if git_project_root:
+            cfg_search_paths.append(git_project_root)
+
+        cfg_paths = (
+            path / Path(filename)
+            for path in cfg_search_paths
+            for filename in defaults.config_files
+        )
 
-    cfg_paths = (
-        path / Path(filename)
-        for path in cfg_search_paths
-        for filename in defaults.config_files
-    )
     for filename in cfg_paths:
-        if not filename.exists():
+        if filepath is not None and not filename.exists():
+            raise ConfigFileNotFound()
```

Comment:
I feel like we probably would want to move it to the front of the method. Otherwise, we need to do multiple unnecessary check when filepath is not provided

Reply:
Got it, I've changed this line before assign `cfg_paths` generators.